install_pim/manual/event_api.rst

```diff
@@ -0,0 +1,219 @@
+
+.. code-block:: bash
+    :linenos:
+
+    $ /path/to/php /path/to/your/pim/bin/console messenger:consume webhook --env=prod
+
+Logs
+----
+
+The consumer writes logs to the standard output.
+It's your responsibility to choose where to write the logs.
+For example, to write in the file ``/tmp/consumer.log``:
+
+.. code-block:: bash
+    :linenos:
+
+    $ /path/to/php /path/to/your/pim/bin/console messenger:consume webhook --env=prod >/tmp/consumer.log 2>&1
+
+Do note that you should ensure the log rotation as well.
+
+Option #1 - Supervisor
+----------------------
+
+It's strongly recommended to use a Process Control System to launch the consumer in production.
+This is not useful in development though.
+
+In this documentation, we will describe how to configure the Process Control System `supervisor <https://github.com/Supervisor/supervisor>`_, to run the consumer.
+These instructions are valid for Debian 9 and Ubuntu 16.04.
+
+Installing supervisor
+**********************
+
+Install `supervisor`:
+
+.. code-block:: bash
+    :linenos:
+
+    $ apt update
+    $ apt install supervisor
+
+For the other platforms, you can follow the install section of the `official documentation <https://github.com/Supervisor/supervisor#documentation>`_.
+
+Configuring supervisor
+**********************
+
+Create a file in the configuration directory of supervisor ``/etc/supervisor/conf.d``.
+
+.. code-block:: bash
+    :linenos:
+
+    [program:akeneo_events_api_consumer]
+    command=/path/to/php /path/to/your/pim/bin/console messenger:consume webhook --env=prod
+    autostart=false
+    autorestart=true
+    stderr_logfile=/var/log/akeneo_events_api_consumer.err.log
+    stdout_logfile=/var/log/akeneo_events_api_consumer.out.log
+    user=my_user
+
+The user ``my_user`` should be the same as the user to run PHP-FPM.
+
+Then, bring the changes into effect:
+
+.. code-block:: bash
+    :linenos:
+
+    $ supervisorctl reread
+    $ supervisorctl update
+
+Launch the daemon
+*****************
+
+.. code-block:: bash
+    :linenos:
+
+    $ supervisorctl start akeneo_events_api_consumer
+
+Option #2 - systemd
+-------------------
+
+If you prefer, you can use ``systemd``, which will also allow to have logs management and auto restart in case of failure.
+
+Configuration files
+*******************
+
+Create ``/etc/systemd/system/akeneo_events_api_consumer@.service``:
+
+.. code-block:: ini
+    :linenos:
+
+    [Unit]
+    Description=Akeneo PIM Events API consumer
+
+    [Service]
+    Type=forking
+    User=root
+    WorkingDirectory=/path/to/home/user/.systemd
+    ExecStart=/usr/local/bin/akeneo_events_api_consumer_launcher.sh
+    After=apache2.service
+    Restart=always
+
+    [Install]
+    WantedBy=multi-user.target
+
+Create ``/usr/local/bin/akeneo_events_api_consumer_launcher.sh``:
+
+.. code-block:: bash
+    :linenos:
+
+    su -c "/path/to/akeneo/bin/console messenger:consume webhook --env=prod &" akeneo
+    exit 0
```

Comment:
We need to check if this sh file is still necessary.

Reply:
Checked in https://github.com/akeneo/pim-docs/pull/1560/commits/c7a35fef6f0edcb52f15bc658c19d3b47fc843dc